pennylane/devices/qubit_mixed/simulate.py

```diff
@@ -0,0 +1,187 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Simulate a quantum script for a qubit mixed state device."""
+# pylint: skip-file
+# black: skip-file
+import pennylane as qml
+from pennylane.typing import Result
+
+from .apply_operation import apply_operation
+from .initialize_state import create_initial_state
+from .measure import measure
+
+INTERFACE_TO_LIKE = {
```

Comment:
Not sure if this is necessary. The interface should've already been resolved at the `QNode`.

https://github.com/PennyLaneAI/pennylane/blob/9a43c7bb9f1cf10516b8932d55d4642b42628db9/pennylane/workflow/qnode.py#L579

Reply:
I rm it but seemingly many other places in tests used interface = None. Therefore I added another branch in `get_final_state` to resolve such issues but effectively this means I didn't really rm this from the interface dict😅also the same `None` entry appears in module `qubit` and `qutrit_mixed` as well so I guess once again this remains as some future improvment story